core/app/models/spree/credit_card.rb

```diff
@@ -97,8 +105,10 @@ def set_last_digits
     # @return [String] the credit card type if it can be determined from the
     #   number, otherwise the empty string
     def try_type_from_number
-      numbers = number.delete(' ') if number
-      CARD_TYPES.find{ |type, pattern| return type.to_s if numbers =~ pattern }.to_s
+      CARD_TYPES.each do |type, pattern|
```

Comment:
Oh, nasty. You have the early return because you return empty string below if a number matches.

I would prefer `.select {...} || ''`, but it's on you to decide